utils/utils.go

```diff
@@ -382,3 +382,17 @@ func Sync(path string) error {
 	}
 	return nil
 }
+
+// Syncfs ensures the file system at path is synced to disk
+func Syncfs(path string) error {
+	f, err := os.OpenFile(path, os.O_RDONLY, 0o755)
```

Comment:
nit: this can be simplified to `os.Open(path)`.

Alternatively, we can use `unix.Open(path, unix.O_PATH|unix.O_CLOEXEC|unix.O_DIRECTORY, 0)` to get the fd directly. Doesn't really matter though.